pkg/scheduler/framework/plugins/noderesources/resource_allocation.go

```diff
@@ -35,14 +37,43 @@ type resourceAllocationScorer struct {
 	Name string
 	// used to decide whether to use Requested or NonZeroRequested for
 	// cpu and memory.
-	useRequested        bool
-	scorer              func(requested, allocable resourceToValueMap) int64
-	resourceToWeightMap resourceToWeightMap
+	useRequested           bool
+	scorer                 func(requested, allocatable resourceToValueMap) int64
+	resourceToWeightMap    resourceToWeightMap
+	resourceToValueMapPool *sync.Pool
```

Comment:
Is this possible to reduce the efficiency of calculating score in parallel because of the race lock?